include/hcl/concurrent/unordered_map/unordered_map.h

```diff
@@ -0,0 +1,301 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Distributed under BSD 3-Clause license.                                   *
+ * Copyright by The HDF Group.                                               *
+ * Copyright by the Illinois Institute of Technology.                        *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of Hermes. The full Hermes copyright notice, including  *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the top directory. If you do not  *
+ * have access to the file, you may request a copy from help@hdfgroup.org.   *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#ifndef INCLUDE_HCL_UNORDERED_MAP_CONCURRENT_H_
+#define INCLUDE_HCL_UNORDERED_MAP_CONCURRENT_H_
+
+/**
+ * Include Headers
+ */
+
+#include <hcl/common/debug.h>
+#include <hcl/common/singleton.h>
+#include <hcl/communication/rpc_factory.h>
+#include <hcl/communication/rpc_lib.h>
+/** MPI Headers**/
+#include <mpi.h>
+/** RPC Lib Headers**/
+#ifdef HCL_ENABLE_RPCLIB
+
+#include <rpc/client.h>
+#include <rpc/rpc_error.h>
+#include <rpc/server.h>
+
+#endif
+/** Thallium Headers **/
+#if defined(HCL_ENABLE_THALLIUM_TCP) || defined(HCL_ENABLE_THALLIUM_ROCE)
+#include <thallium.hpp>
+#endif
+
+/** Boost Headers **/
+#include <boost/algorithm/string.hpp>
+/** Standard C++ Headers**/
+#include <hcl/common/container.h>
+
+#include <functional>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include <float.h>
+#include "Block_chain_memory.h"
+
+namespace hcl {
+
+template <class KeyT, 
+	  class ValueT,
+	  class HashFcn=std::hash<KeyT>,
+	  class EqualFcn=std::equal_to<KeyT>>
+class unordered_map_concurrent : public container 
```

Comment:
Fix this naming. You should have lock and lock-free folders and they should have unordered_map as a sub-folder